pkg/apis/azuredisk/v1alpha1/types.go

```diff
@@ -25,21 +25,31 @@ import (
 
 // AzVolume is a specification for a AzVolume resource
 type AzVolume struct {
+	// spec defines the desired state of a AzVolumeAttachment.
+	// Required.
 	metav1.TypeMeta   `json:",inline"`
 	metav1.ObjectMeta `json:"metadata,omitempty"`
 
+	// status represents the current state of AzVolumeAttachment.
+	// Nil status indicates that the underlying volume of AzVolumeAttachment has not yet been attached to the specified node
+	// +optional
 	Spec   AzVolumeSpec   `json:"spec"`
-	Status AzVolumeStatus `json:"status"`
+	Status AzVolumeStatus `json:"status,omitempty"`
```

Comment:
`Status` should be marked optional and made a pointer, i.e. `Status *AzVolumeStatus`.